client/galaxy/scripts/mvc/tool/tool-form.js

```diff
@@ -8,42 +8,64 @@ define([ 'utils/utils', 'mvc/ui/ui-misc', 'mvc/ui/ui-modal', 'mvc/tool/tool-form
-                            onclick  : function() {
-                                execute_btn.wait();
-                                form.portlet.disable();
-                                self.submit( options, function() {
-                                    execute_btn.unwait();
-                                    form.portlet.enable();
-                                } );
-                            }
-                        })
-                    }
-                    // remap feature
-                    if ( options.job_id && options.job_remap ) {
-                        options.inputs.push({
-                            label       : 'Resume dependencies from this job',
-                            name        : 'rerun_remap_job_id',
-                            type        : 'select',
-                            display     : 'radio',
-                            ignore      : '__ignore__',
-                            value       : '__ignore__',
-                            options     : [ [ 'Yes', options.job_id ], [ 'No', '__ignore__' ] ],
-                            help        : 'The previous run of this tool failed and other tools were waiting for it to finish successfully. Use this option to resume those tools using the new output(s) of this tool run.'
-                        });
+
+                    // build request url
+                    var build_url = '';
+                    var build_data = {};
+                    var job_id = options.job_id;
+                    if ( job_id ) {
+                        build_url = Galaxy.root + 'api/jobs/' + job_id + '/build_for_rerun';
+                    } else {
+                        build_url = Galaxy.root + 'api/tools/' + options.id + '/build';
+                        build_data = $.extend( {}, Galaxy.params );
+                        build_data[ 'tool_id' ] && ( delete build_data[ 'tool_id' ] );
                     }
+                    options.version && ( build_data[ 'tool_version' ] = options.version );
+
+                    // get initial model
+                    Utils.get({
+                        url     : build_url,
+                        data    : build_data,
+                        success : function( data ) {
+                            if( !data.display ) {
+                                window.location = Galaxy.root;
+                                return;
+                            }
+                            form.model.set( data );
+                            self._customize( form );
+                            Galaxy.emit.debug('tool-form-base::_buildModel()', 'Initial tool model ready.', data);
+                            process.resolve();
+                        },
+                        error   : function( response, status ) {
+                            var error_message = ( response && response.err_msg ) || 'Uncaught error.';
+                            if ( status == 401 ) {
```

Comment:
This is not new code, and it's not a problem with this PR, but is this something we can handle elsewhere down the road so we don't have to augment all these requests with login redirects?